app/models/project.rb

```diff
@@ -56,9 +56,8 @@ def available_extensions
   end
 
   def extensions
-    attributes["extensions"] ||= []
-    attributes["extensions"].reject!(&:blank?)
-    attributes["extensions"]
+    attributes["extensions"].reject!(&:blank?) if attributes["extensions"]
+    attributes["extensions"] || []
```

Comment:
You mean `||=` on line 60. As I mentioned before, you should also invert these lines, and leave in line 61...then you don't need the `if attributes["extensions"]`.

(Also, feel free to add a `return` in line 61. I know it's like nails on chalkboard to Ruby developers, but it really does make your code much clearer.)